### PR TITLE
Feat: Review lambda authoriser

### PR DESCRIPTION
**Pull request by lizacullis**
Have tested that this works with the test `secret`, `signature` and `payload`.  Photos below.
But might need to test full flow with Github Webhooks once app is set up. 

**Postman request with test signature and payload:** 
Screenshot 2023-08-11 at 16 40 13

**Authoriser approved:** 
Screenshot 2023-08-11 at 16 15 31

### CODE REVIEW

File: services/core/functions/review-lambda/auth.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  // Verify Token
+  const sigHex = header.split("=")[1];
+  const sigBytes = hexToBytes(sigHex);
+  const dataBytes = encoder.encode(payload);
+  const equal = await subtle.verify(algorithm.name, key, sigBytes, dataBytes);
+
+  return equal;
+};
+
+function hexToBytes(hex: string) {
```
Comment (mattzcarey) [inline services/core/functions/review-lambda/auth.ts:32]: I'd prefer arrow syntax here.

File: services/core/functions/review-lambda/config.ts [hunk 2]
```diff
@@ -20,6 +21,7 @@ export class ReviewLambda extends NodejsFunction {
       architecture: Architecture.ARM_64,
       environment: {
         OPENAI_API_KEY_PARAM_NAME: OPENAI_API_KEY_PARAM_NAME,
+        GITHUB_SECRET_PARAM_NAME: GITHUB_SECRET_PARAM_NAME,
         LANGCHAIN_API_KEY_PARAM_NAME: LANGCHAIN_API_KEY_PARAM_NAME,
         LANGCHAIN_TRACING_V2: "true",
         LANGCHAIN_PROJECT: "code-review-gpt",
```
Comment (mattzcarey) [inline services/core/functions/review-lambda/config.ts:27]: change this to 'review' please

**Review by gowoons — APPROVED** [review]
Looks good !

File: services/core/functions/review-lambda/index.ts [hunk 3]
```diff
@@ -12,6 +14,29 @@ interface ReviewLambdasBody {
...
+    });
+  }
+
+  const authenticated = await authenticate(header, event.body);
+  if (!authenticated) {
+    return Promise.resolve({
+      statusCode: 401,
+      body: "Unauthorized.",
+    });
+  }
```
Comment (mattzcarey) [inline services/core/functions/review-lambda/index.ts:38]: We should build out some generic error responses and codes but this is fine for now.

**Review by mattzcarey — APPROVED** [review]
Approved but please change the style thing.

**Pull request merged**